packages/docs/versioned_docs/version-2.0.0-alpha-3/protocol/06-trees.md

```diff
@@ -4,29 +4,31 @@ description: The tree structures that are used in UniRep protocol.
 
 # Trees
 
-Each attester has a separate version of these trees each epoch.
+Each attester has an updated version of each tree for each epoch.
 
 ## **State tree**
 
-* A state tree stores the updated user state after a user signs up and a user performs the [user state transition](05-user-state-transition.md).
-* It is an **incremental merkle tree** with it's leaves storing users' `identityNullifier`s and starting reputation, e.g.
+* A state tree stores the updated user state after a user signs up and after a [user state transition](05-user-state-transition.md) is performed.
+* This is an **incremental merkle tree**, with it's leaves storing users' `identityNullifier`s and starting reputation, e.g.
   * a state tree leaf: `hash(identityNullifier, attesterId, epoch, posRep, negRep, graffiti, timestamp)`
   * The default leaf is `0`
 
 :::info
 **NOTE:** this is an incremental merkle tree so leaves are inserted from left (leaf index 0) to right, one by one, instead of inserted directly into the specified leaf index.
 
-**NOTE:** since state tree leaf is the hash of `identityNullifier` and other values, observers will be not be able to tell which user has inserted leaves into the tree.
+**NOTE:** since state tree leaf is the hash of `identityNullifier` and other values, observers are not able to determine which user has inserted leaves into the tree.
 :::
 
 ## **Epoch tree**
 
 * An epoch tree is used to **track the reputation received by epoch keys**. Non-repudiability is enforced at the circuit and smart contract level.
 
-* An epoch tree is a **sparse merkle tree** with its leaves storing the hash of reputation received by the epoch key, e.g.,
+* This is a **sparse merkle tree**, with its leaves storing the hash of reputation received by the epoch key, e.g.,
   * leaf index: epoch key
   * leaf value: `H(posRep, negRep, graffiti, timestamp)`
 
 :::info
 See also: [Reputation](04-reputation.md)
 :::
+
+Read more about Merkle Trees here [(https://medium.com/@kelvinfichter/whats-a-sparse-merkle-tree-acda70aeb837)]
```

Comment:
Same question here about linking out to supporting content.

Reply:
I think it's fine to link. Just need to review it first.